account_invoice_qr_code_sepa_payconiq/models/res_partner_bank.py

```diff
@@ -0,0 +1,182 @@
+
+    def _get_qr_code_generation_params(
+        self,
+        qr_method,
+        amount,
+        currency,
+        debtor_partner,
+        free_communication,
+        structured_communication,
+    ):
+        """
+        https://developer.payconiq.com/online-payments-dock/#creating-the-payconiq-qr-code77
+
+        """
+        if qr_method == "payconiq_qr":
+            return {
+                "payconiq_qr": True,
+                "D": "Payment",
+                "A": self._get_payconiq_qr_amount(amount),
+                "R": structured_communication
+                if structured_communication
+                else free_communication,
+            }
+        return super()._get_qr_code_generation_params(
+            qr_method,
+            amount,
+            currency,
+            debtor_partner,
+            free_communication,
+            structured_communication,
+        )
+
+    def _get_qr_code_url(
+        self,
+        qr_method,
+        amount,
+        currency,
+        debtor_partner,
+        free_communication,
+        structured_communication,
+    ):
+        params = self._get_qr_code_generation_params(
+            qr_method,
+            amount,
+            currency,
+            debtor_partner,
+            free_communication,
+            structured_communication,
+        )
+        if params and params.get("payconiq", False):
+            params.pops("payconiq_qr")
+            return urllib.parse.urlencode(params)
+        return super()._get_qr_code_url(
+            qr_method,
+            amount,
+            currency,
+            debtor_partner,
+            free_communication,
+            structured_communication,
+        )
+
+    def _eligible_for_qr_code(self, qr_method, debtor_partner, currency):
+        if qr_method == "payconiq_qr":
+            return (
+                currency.name == "EUR"
+                and self.acc_type == "iban"
+                and self.sanitized_acc_number[:2] in ["LU"]
+            )
+        return super()._eligible_for_qr_code(qr_method, debtor_partner, currency)
+
+    def _get_qr_code_base64(
+        self,
+        qr_method,
+        amount,
+        currency,
+        debtor_partner,
+        free_communication,
+        structured_communication,
+    ):
+        """
+        The url to get QR code image from Payconiq is composed by several parameters:
+            - The type of the image
+            - The image size
+            - The redirection url (to launch the payconiq application - this is the url
+              inside the QR code itself)
+        """
+        params = self._get_qr_code_generation_params(
+            qr_method,
+            amount,
+            currency,
+            debtor_partner,
+            free_communication,
+            structured_communication,
+        )
+        if params and params.get("payconiq_qr", False):
+            params.pop("payconiq_qr")
```

Comment:
why ?

Reply:
Because Odoo needs this to detect the payment type but payconiq not.

Reply:
OK. You can do something like ```if params and params.pop("payconiq_qr", False):```